lib/identity_cache/cached/belongs_to.rb

```diff
@@ -60,14 +62,16 @@ def fetch(records)
               hash[parent_id] = child_record
             end
           end
-          parent_records = reflection.klass.fetch_multi(ids_to_child_record.keys)
-          parent_records.each do |parent_record|
-            child_record = ids_to_child_record[parent_record.id]
-            child_record.instance_variable_set(records_variable_name, parent_record)
+
+          load_strategy.load_multi(reflection.klass.cached_primary_index, ids_to_child_record.keys) do |parent_records_by_id|
+            parent_records_by_id.each do |id, parent_record|
+              next unless child_record = ids_to_child_record[id]
```

Comment:
When would `child_record` be `nil`?  Isn't it the owner of the association?  I don't see why we would have loaded a record without an owner.

I think we should be able to do

```suggestion
             child_record = ids_to_child_record.fetch(id)
```

We also have `next unless` in the polymorphic case above, which also seems like it should be unnecessary.

It looks like you also repeated this in reference/has_many.rb and reference/has_one.rb as well.

Reply:
There's a nil value in my prefetch test when prefetching multiple items at different levels. 🤔